beacon-chain/cache/depositcache/deposits_cache.go

```diff
@@ -61,9 +61,17 @@ type DepositCache struct {
 
 // NewDepositCache instantiates a new deposit cache
 func NewDepositCache() *DepositCache {
+	finalizedDepositsTrie, err := trieutil.NewTrie(int(params.BeaconConfig().DepositContractTreeDepth))
+	if err != nil {
+		panic(err)
```

Comment:
No panic please, return an error if needed

Reply:
I based my solution on this:

https://github.com/prysmaticlabs/prysm/blob/c2615168d9af98c6c1a348b943b43dc158d57fbf/beacon-chain/cache/hot_state_cache.go#L33-L37

I will change my constructor's return type to `(*DepositCache, error)`.

Reply:
That should be return an errors. In fact for all the instantiations of caches